enhancements/etcd/automated-backups.md

```diff
@@ -0,0 +1,284 @@
+---
+title: automated-backups-of-etcd
+authors:
+  - "@hasbro17"
+reviewers:
+  - "@deads2k"
+  - "@dusk125"
+  - "@Elbehery"
+  - "@tjungblu"
+  - "@williamcaban"
+approvers:
+  - "@deads2k"
+  - "@williamcaban"
+api-approvers:
+  - "@deads2k"
+creation-date: 2023-03-17
+last-updated: 2023-03-17
+tracking-link:
+  - https://issues.redhat.com/browse/ETCD-81
+see-also:
+  - "https://issues.redhat.com/browse/OCPBU-252"
+  - "https://issues.redhat.com/browse/OCPBU-160"
+---
+
+
+# Automated Backups of etcd
+
+## Summary
+
+Enable the automated backups of etcd snapshots and other metadata necessary to restore an openshift cluster from a quorum loss scenario.
+
+## Motivation
+
+The [current documented procedure](https://docs.openshift.com/container-platform/4.12/backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.html) for performing a backup of an OpenShift
+cluster is manually initiated. This reduces the likelihood of a timely backup being available during a disaster recovery scenario.
+
+The procedure also requires gaining a root shell on a control plane
+node. Shell access to OpenShift control plane nodes access is generally
+discouraged due to the potential for affecting the reliability of the
+node.
+
+### Goals
+
+- One-time cluster backup can be triggered without requiring a root shell
+- Scheduled backups can be configured after cluster installation
+- Backups are saved locally on the host filesystem of control-plane nodes
```

Comment:
How does this work for hosted control planes?
1. can the admin in the hosted cluster, who doesn't manage the control plane, initiate this backup?
2. if they can what protections are in place to ensure backups do not impact the hosting node in a way that may negaively impact other control plane pods hosted on the node where the backup is saved?

Reply:
"Separate ownership" today, doesn't mean it needs to be separate forever, does it?  It seems like it would be possible to move towards a unified approach.  For example, in a hosted control plane world, it's not clear to me [which persona(s)][1] would have which access to backups.  Possibly the Cluster Service Consumer, if we lump backup restoration in as similar to "drive upgrades".  Or possibly the Cluster Service Provider, if we lump backup restoration in under "Has cluster admin management".  But either way, I'd expect etcd backup/restore tooling to live on the management cluster and not leak out to the `Cluster Instance ...` roles.  Which would argue for [a namespaced CRD][2], so a management cluster could have these resources in each of its per-hosted-cluster namespaces.

[1]: https://hypershift-docs.netlify.app/reference/concepts-and-personas/#personas
[2]: https://github.com/openshift/enhancements/pull/1370#discussion_r1163950907

Reply:
interesting take, I would rather see more separation instead of more unification. For example, we rebased etcd earlier this year and discovered it broke a customer cluster. checkout https://github.com/etcd-io/etcd/issues/15062

We didn't even know that Hypershift was using openshift/etcd as their etcd. They are certainly not using CEO, which makes reusing the backup CRD moot as they would have to implement their own business logic for backups anyway. 

CEO is also currently a singleton in the cluster (don't think we want to change that either), so namespacing that CRD would make little sense in OCP.